packages/dds/tree/src/shared-tree/schematizingTreeView.ts

```diff
@@ -87,6 +87,21 @@ export class SchematizingSimpleTreeView<
 	in out TRootSchema extends ImplicitFieldSchema | UnsafeUnknownSchema,
 > implements TreeViewAlpha<TRootSchema>, WithBreakable
 {
+	/**
+	 * Optional user-provided label associated with the transaction as a commit is being applied, if provided.
+	 * This value is intended to be read within event handlers like `commitApplied`.
+	 * This value is cleared after each transaction to prevent providing stale/incorrect labels.
+	 */
+	private static _currentTransactionLabel: unknown | undefined;
```

Comment:
I think this property might work out even better if you put it on `TreeCheckout` - and it could be a member rather than be static. I know we discussed using a static because it's correct and it's easy - but if a non-static member is _just as easy_, then non-static is probably better style/convention.